src/routes/auth/auth.service.ts

```diff
@@ -58,6 +68,26 @@ export const authService = {
 
     return { ...dbResponse, token }
   },
+  // TODO: zod for token & email?
+  async verifyEmail(email: string, token: string): Promise<void> {
+    const payload = await verify(token, process.env.JWT_SECRET!)
+
+    if (!payload || payload.email !== email) {
+      throw appError('auth/invalid-token', StatusCodes.UNAUTHORIZED)
+    }
+    const user = await isUserExist(email)
+
+    if (!user) {
+      throw appError('auth/user-not-found', StatusCodes.NOT_FOUND)
+    }
+    const verified = await isUserVerified(email)
```

Comment:
It looks like `isUserExist` & `isUserVerified` are 2 calls to DB - can we think about one?